src/overlays/actors/ovl_Bg_Haka_Zou/z_bg_haka_zou.c

```diff
@@ -30,35 +56,344 @@ const ActorInit Bg_Haka_Zou_InitVars = {
+    }
+}
+
+void func_80883328(BgHakaZou* this, GlobalContext* globalCtx) {
+    Vec3f effectPos;
+    s32 i;
+    s32 j;
+
+    this->dyna.actor.posRot.rot.x += this->dyna.actor.posRot.rot.x * 0.125f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Zou/func_808828F4.s")
+    if (Math_ScaledStepToS(&this->dyna.actor.shape.rot.x, -0x4000, this->dyna.actor.posRot.rot.x) != 0) {
+        effectPos.x = this->dyna.actor.posRot.pos.x;
+        effectPos.y = this->dyna.actor.posRot.pos.y;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Zou/func_80882A70.s")
+        for (j = 0; j < 2; effectPos.x -= 112.0f, j++) {
+            effectPos.z = this->dyna.actor.posRot.pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Zou/func_80882BDC.s")
+            for (i = 0; i < 4; i++) {
+                effectPos.z -= (i == 2) ? 550.0f : 50.0f;
+                func_800286CC(globalCtx, &effectPos, &sEffectVec, &sEffectVec, (Rand_ZeroOne() * 200.0f) + 1000.0f,
+                              200);
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Zou/func_80882CC4.s")
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_STONE_BOUND);
+        this->timer = 25;
+        this->actionFunc = func_808834D8;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Zou/func_80882E54.s")
+void func_808834D8(BgHakaZou* this, GlobalContext* globalCtx) {
+    f32 moveDist;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Zou/func_80883000.s")
+    if (this->timer != 0) {
+        this->timer--;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Zou/func_80883104.s")
+    moveDist = ((this->timer % 2) ? 15.0f : -15.0f);
+    this->dyna.actor.posRot.pos.y += ((this->timer & 0xFE) * 0.04f * moveDist);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Zou/func_80883144.s")
+    if (this->timer == 0) {
+        this->actionFunc = BgHakaZou_Do_Nothing;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Zou/func_80883254.s")
+void BgHakaZou_Do_Nothing(BgHakaZou* this, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Zou/func_80883328.s")
+void BgHakaZou_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgHakaZou* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Zou/func_808834D8.s")
+    this->actionFunc(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Zou/func_80883568.s")
+    if (this->dyna.actor.params == 3) {
+        Actor_MoveForward(&this->dyna.actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Zou/BgHakaZou_Update.s")
+void BgHakaZou_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    static Gfx* displayLists[] = { 0x060064E0, 0x06005CE0, 0x06000A10, 0x06005CE0 };
```

Comment:
I suggest leaving that array right above the function, unless it has to be in-function static to match. It could get pretty verbose with the full symbols. Also suggest including a trailing comma because it will very likely need it once it has full symbols.